tests/agent/test_docker_agent.py

```diff
@@ -488,28 +494,40 @@ def test_docker_agent_deploy_flow_run_config(api, run_kind, has_docker_storage):
         host_config = {}
         exp_host_config = {
             "auto_remove": True,
-            "extra_hosts": {"host.docker.internal": "host-gateway"},
         }
         run = None if run_kind == "missing" else UniversalRun()
 
+    if docker_engine_version == "20.10.0":
+        exp_host_config["extra_hosts"] = {"host.docker.internal": "host-gateway"}
+        warns_on_old_version = nullcontext()
+    else:
+        warns_on_old_version = pytest.warns(
```

Comment:
Conditional logic within a test that actually splits into different asserts is in general brittle; it's very easy to update the test / a fixture and not actually test both paths without ever realizing it.  In this case, the solution would be to preserve `test_docker_agent_deploy_flow_run_config` as the happy path, and introduce a new test `test_docker_agent_warns_on_old_docker_versions` that tests for the warning.

Reply:
Sounds good to me